sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -297,46 +277,83 @@ public boolean isRetryAfter(final @NotNull String itemType) {
-      connection.getInputStream().close();
       options.getLogger().log(DEBUG, "Envelope sent successfully.");
       return TransportResult.success();
     } catch (IOException e) {
-      try {
-        responseCode = connection.getResponseCode();
+      final TransportResult errorResult = getResultFromErrorCode(connection, e);
+      responseCode = errorResult.getResponseCode();
+      return errorResult;
+    } catch (Exception e) {
+      return getDefaultErrorResult(e);
+    } finally {
+      updateRetryAfterLimits(connection, responseCode);
 
-        // TODO: 403 part of the protocol?
-        if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
-          options
-              .getLogger()
-              .log(DEBUG, "Envelope was rejected by the Sentry server due to a filter.");
-        }
-        logErrorInPayload(connection);
-        return TransportResult.error(responseCode);
-      } catch (IOException responseCodeException) {
-        // this should not stop us from continuing. We'll just use -1 as response code.
-        options
-            .getLogger()
-            .log(WARNING, "Failed to obtain response code while analyzing event send failure.", e);
-      }
+      closeAndDisconnectConnection(connection);
+    }
+  }
+
+  /**
+   * Closes the Response stream and disconnect the connection
+   *
+   * @param connection the HttpURLConnection
+   */
+  private void closeAndDisconnectConnection(final @NotNull HttpURLConnection connection) {
+    try {
+      connection.getInputStream().close();
+    } catch (IOException e) {
+      options.getLogger().log(ERROR, e, "Error while closing the connection.");
+    }
+    connection.disconnect();
+  }
+
+  /**
+   * Returns a TransportResult error with the given responseCode
+   *
+   * @param connection the HttpURLConnection
+   * @param ioException the IOException
+   * @return the TransportResult error
+   */
+  private @NotNull TransportResult getResultFromErrorCode(
+      final @NotNull HttpURLConnection connection, final @NotNull IOException ioException) {
+    try {
+      final int responseCode = connection.getResponseCode();
 
       logErrorInPayload(connection);
       return TransportResult.error(responseCode);
-    } catch (Exception e) {
-      options
-          .getLogger()
-          .log(WARNING, "Failed to obtain error message while analyzing event send failure.", e);
-      return TransportResult.error(-1);
-    } finally {
-      updateRetryAfterLimits(connection, responseCode);
+    } catch (IOException responseCodeException) {
+      options.getLogger().log(ERROR, responseCodeException, "Error getting responseCode");
 
-      connection.disconnect();
+      // this should not stop us from continuing.
+      return getDefaultErrorResult(ioException);
     }
   }
 
+  /**
+   * Logs a default error message and return the TransportResult error with -1 responseCode
+   *
+   * @param exception the Exception
+   * @return the TransportResult error
+   */
```

Comment:
`l`: It' really nice that we have all methods commented, but if they are private and do simple things, I don't think they are needed, because the comment doesn’t add any value IMO. [Detekt](https://arturbosch.github.io/detekt/comments.html#commentoverprivatefunction) has even a rule against this.

Reply:
this would make sense for closed source, but I'm highly against for open-source.
comments don't add an extra size to the final bytecode and it helps a lot for contributors.

Reply:
I disagree, but that's fine, because I don't have a strict opinion on this. Maybe we can discuss this once f2f. Not worth discussing it right now. 😄

Reply:
sure, let's discuss offline.